src/main/java/org/jenkinsci/plugins/saml/user/LoginDetailsProperty.java

```diff
@@ -0,0 +1,172 @@
+import java.util.logging.Logger;
+
+/**
+ * Store details about create and login processes
+ *
+ * @author Kuisathaverat
+ */
+public class LoginDetailsProperty extends UserProperty {
+    private static final Logger LOG = Logger.getLogger(LoginDetailsProperty.class.getName());
+    private static final String ISO_8601 = "yyyy-MM-dd'T'HH:mm:ssZ";
+    private long createTimestamp;
+    private long lastLoginTimestamp;
+    private long loginCount;
+
+    
+    @DataBoundConstructor
+    public LoginDetailsProperty() {
+    }
+
+    public static LoginDetailsProperty currentUserLoginDetails(){
+        User user = User.current();
+        LoginDetailsProperty loginDetails = null;
+        if(user!=null && user.getProperty(LoginDetailsProperty.class) != null){
+            loginDetails = user.getProperty(LoginDetailsProperty.class);
+        }
+        return loginDetails;
+    }
+
+    public static void currentUserSetLoginDetails(){
+        User user = User.current();
+        if(user!=null && user.getProperty(LoginDetailsProperty.class) != null){
+            LoginDetailsProperty loginDetails = user.getProperty(LoginDetailsProperty.class);
+            loginDetails.update();
+        }
+    }
+
+    public void update(){
+        long now = System.currentTimeMillis();
+        if(getCreateTimestamp() == 0){
+            setCreateTimestamp(now);
+        }
+
+        setLastLoginTimestamp(now);
+        setLoginCount(getLoginCount()+1);
+        try {
+            user.save();
+        } catch (java.io.IOException e) {
+            LOG.log(Level.WARNING,e.getMessage(),e);
+        }
+    }
+
+    public long getCreateTimestamp() {
+        return createTimestamp;
+    }
+
+    public long getLastLoginTimestamp() {
+        return lastLoginTimestamp;
+    }
+
+    public String getCreateDate() {
+        return FastDateFormat.getInstance(ISO_8601).format(new Date(createTimestamp));
+    }
+
+    public String getLastLoginDate() {
+        return FastDateFormat.getInstance(ISO_8601).format(new Date(lastLoginTimestamp));
+    }
+
+    public long getLoginCount() {
+        return loginCount;
+    }
+
+    public void setCreateTimestamp(long createTimestamp) {
+        this.createTimestamp = createTimestamp;
+    }
+
+    public void setLastLoginTimestamp(long lastLoginTimestamp) {
+        this.lastLoginTimestamp = lastLoginTimestamp;
+    }
+
+    public void setLoginCount(long loginCount) {
+        this.loginCount = loginCount;
+    }
+
+    @Override
+    public UserProperty reconfigure(StaplerRequest req, JSONObject form) throws FormException {
+        return this;
+    }
+
+
+    /**
+     * Listen to the login success/failure event to persist {@link GrantedAuthority}s properly.
+     */
+    @Extension
+    public static class SecurityListenerImpl extends jenkins.security.SecurityListener {
+        @Override
+        protected void authenticated(@javax.annotation.Nonnull org.acegisecurity.userdetails.UserDetails details) {
+        }
```

Comment:
Huh, wonder why I made all these methods `abstract`. Should have left default no-op implementations.